### PR TITLE
Refactor CAPTCHA extracting Driver

### PR DESCRIPTION
**Pull request by klimov-paul**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | yes
| Tests pass?   | yes
| Fixed issues  | #13799

### CODE REVIEW

**Comment by vladis84** [comment]
Will you write tests?

File: framework/captcha/Driver.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ * By configuring the properties of Driver, you may customize the appearance of
+ * the generated CAPTCHA images, such as the font color, the background color, etc.
+ *
+ * @author Paul Klimov <klimov.paul@gmail.com>
+ * @author Qiang Xue <qiang.xue@gmail.com>
+ * @since 2.1.0
+ */
+abstract class Driver extends Component implements DriverInterface
+{
+    use VerifyCodeGeneratorTrait;
```
Comment (vladis84) [inline framework/captcha/Driver.php]: Trait not needed. Used once.

Comment (klimov-paul) [inline framework/captcha/Driver.php]: Trait might ve needed for particular developer-defined driver, which renders image with other options or format, but wishes to use existing code generation.

**Comment by vladis84** [comment]
How can I change the driver?

File: framework/captcha/ImagickDriver.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+class ImagickDriver extends Driver
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function init()
+    {
+        parent::init();
+
+        if (!extension_loaded('imagick') || !in_array('PNG', (new \Imagick())->queryFormats('PNG'), true)) {
```
Comment (vladis84) [inline framework/captcha/ImagickDriver.php]: If `imagick` not load, the \Imagick throw error

**Comment by samdark** [comment]
@vladis84 I guess because there's no changelog yet.

**Review by samdark — APPROVED** [review]

**Pull request merged**